hummingbot/market/market_base.pyx

```diff
@@ -129,11 +129,11 @@ cdef class MarketBase(NetworkIterator):
     async def cancel_all(self, timeout_seconds: float) -> List[CancellationResult]:
         raise NotImplementedError
 
-    cdef str c_buy(self, str trading_pair, object amount, object order_type=OrderType.MARKET,
+    cdef str c_buy(self, str trading_pair, object amount, object order_type=OrderType.LIMIT,
                    object price=s_decimal_NaN, dict kwargs={}):
```

Comment:
Ideally, price can no longer have a default value of NaN, the users of this function needs to supply a price.